sarracen/readers/read_phantom.py

```diff
@@ -181,18 +183,26 @@ def _read_array_blocks(fp, def_int_dtype, def_real_dtype):
         else:
             df = _read_array_block(fp, df, n[i], nums[i], def_int_dtype, def_real_dtype)
 
-    return pd.concat([df, df_sinks], ignore_index=True)
-
+    if drop_sinks:
+        return df
+    else:
+        return pd.concat([df, df_sinks], ignore_index=True)
```

Comment:
I suggest returning both df and df_sinks from this function, and then doing the combining or separating after so it's all done in one place.